src/plugins/es_ui_shared/static/forms/components/fields/text_field.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiFieldText } from '@elastic/eui';
+
+import { FieldHook } from '../../hook_form_lib';
+
+interface Props {
+  field: FieldHook;
+  euiFieldProps?: Record<string, any>;
+  idAria?: string;
+  [key: string]: any;
+}
+
+export const TextField = ({ field, euiFieldProps = {}, ...rest }: Props) => {
+  const isInvalid = !field.isChangingValue && field.errors.length > 0;
+  const errorMessage =
+    !field.isChangingValue && field.errors.length ? (field.errors[0].message as string) : null;
+
+  return (
+    <EuiFormRow
+      label={field.label}
+      helpText={field.helpText}
+      error={errorMessage}
+      isInvalid={isInvalid}
+      fullWidth
+      data-test-subj={rest['data-test-subj']}
+      describedByIds={rest.idAria ? [rest.idAria] : undefined}
+    >
+      <EuiFieldText
+        isInvalid={isInvalid}
+        value={field.value as string}
+        onChange={field.onChange}
+        isLoading={field.isValidating}
+        fullWidth
+        data-test-subj="input"
```

Comment:
I am not thinking of any for now. This would be testing implementation details IMO. I decided to either go with `input` or `select` that should cover all the different form inputs.

In the tests, we will access it with a child selector `"myTestSubj.input"`